### PR TITLE
Fixed: Code to use a single service to subscribe webhook(#2fz59a8)

### PR DESCRIPTION
**Pull request by shashwatbangar**
### Related Issues

Closes #
https://github.com/hotwax/job-manager/commit/5c03c11053f597aaf5391eb566cd0a186f6b8102

### Short Description and Why It's Useful

Now only a single service will be called for subscribe methods.

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/store/modules/webhook/actions.ts [hunk 1]
```diff
@@ -36,30 +36,13 @@ const actions: ActionTree<WebhookState, RootState> = {
...
-      return;
-    }
+    const webhookMethod: any = WebhookService.webhookMethods[id];
 
     let resp;
 
     try {
-      resp = await webhookMethod({ shopifyConfigId: this.state.user.shopifyConfig })
-
+      resp = await webhookMethod({ shopifyConfigId: this.state.user.shopifyConfig }, id)
```
Comment (ymaheshwari1) [inline src/store/modules/webhook/actions.ts:44]: I think we need to call the subscribe webhook action.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/webhook/actions.ts (hunk 1 above)
Comment (shashwatbangar) [reply src/store/modules/webhook/actions.ts]: Done sir!

File: src/store/modules/webhook/actions.ts [hunk 2]
```diff
@@ -36,30 +36,11 @@ const actions: ActionTree<WebhookState, RootState> = {
...
-      return;
-    }
-
     let resp;
 
     try {
-      resp = await webhookMethod({ shopifyConfigId: this.state.user.shopifyConfig })
-
+      resp = await WebhookService.subscribeWebhook({ shopifyConfigId: this.state.user.shopifyConfig }, id)
+
```
Comment (rathoreprashant) [inline src/store/modules/webhook/actions.ts:43]: Unwanted change, please revert

Comment (shashwatbangar) [reply src/store/modules/webhook/actions.ts:43]: Done sir.

File: src/services/WebhookService.ts [hunk 3]
```diff
@@ -8,62 +8,21 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
+  'PAYMENT_STATUS': '',
+  'RETURNS': '',
+  'NEW_PRODUCTS': '',
+  'DELETE_PRODUCTS': 'service/subscribeProductDeleteWebhook',
+  'BULK_OPERATIONS_FINISH': 'service/subscribeFileStatusUpdateWebhook'
+} as any
 
-const subscribeFileStatusUpdateWebhook = async (payload?: any): Promise <any> => {
+const subscribeWebhook = async (payload?: any, id?: string): Promise <any> => {
+  const endpointUrl = webhookMethods[id as string];
```
Comment (adityasharma7) [inline src/services/WebhookService.ts:23]: Handle case when endpoint is not available like this
```
    if (!webhookMethod) {
      showToast(translate("Configuration missing"));
      return;
    }
 ```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/services/WebhookService.ts (hunk 3 above)
Comment (shashwatbangar) [reply src/services/WebhookService.ts:25]: Done sir!

Comment (adityasharma7) [reply src/services/WebhookService.ts:25]: I think we should improve the object name `webhookMethods`

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/services/WebhookService.ts [hunk 4]
```diff
@@ -8,65 +10,29 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
+} as any
+
+const subscribeWebhook = async (payload?: any, id?: string): Promise <any> => {
+  const endpointUrl = webhookEndpointUrls[id as string];
+  if(!endpointUrl) {
+    showToast(translate("Configuration missing"));
+    return;
+  } else {
+    return api ({
+      url: endpointUrl ? endpointUrl : '',
```
[the remark this answers is not in the document]
Comment (shashwatbangar) [reply src/services/WebhookService.ts:31]: Sure sir, fixing it.

File: src/store/modules/webhook/actions.ts [hunk 5]
```diff
@@ -36,29 +36,10 @@ const actions: ActionTree<WebhookState, RootState> = {
...
-    if (!webhookMethod) {
-      showToast(translate("Configuration missing"));
-      return;
-    }
-
     let resp;
 
     try {
-      resp = await webhookMethod({ shopifyConfigId: this.state.user.shopifyConfigId })
+      resp = await WebhookService.subscribeWebhook({ shopifyConfigId: this.state.user.shopifyConfig }, id)
```
Comment (ymaheshwari1) [inline src/store/modules/webhook/actions.ts:42]: Change it to shopifyConfigId
https://github.com/hotwax/job-manager/blob/main/src/store/modules/user/UserState.ts#L6

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — APPROVED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**